### ISSUE TITLE
Improve food editor input

### ISSUE THREAD

**Issue by davidhealey** [davidhealey/waistline#349]
Especially in the diary page when you just want to make a quick change to the serving or quantity, it's clumsy to be in a text box having to manually delete values one character at a time. I think it would be better to open a modal input window with an empty input box and default to numeric input.

**Comment by ZMcK** [comment]
Most other food tracking apps I've seen use a wheel/picker for the primary adjustment and a numeric input as backup. I've used Lose It! for several months now and it works very well. I see Framework7 has a picker tool. If you're interested in using it, I'd work on implementing it.

**Comment by davidhealey** [comment]
Yeah a picker sounds like a good idea. I'll explore some options this weekend and let you know my thoughts. Whatever we go with is needs to be used for editing serving size, number of servings, and possibly nutriment values, although maybe nutriments can remain as is.

**Comment by davidhealey** [comment]
I think for now I'm going to stick with the input boxes. But I'm going to change their behaviour slightly so that when you click them it automatically highlight the text in the box so you can type straight away to replace it. I'm also going to change the styling of the input boxes to make them more clearly defined.

**Comment by mrhebisan** [comment]
Given that I try to keep serving sizes in grams, I'm glad to see you not choose a spinner.  I'm definitely interested in the field auto highlighting.  Maybe even just flat out auto selected when the editor is open?

As is, the margin on the containing editor is too close to the edge of the screen and I often cannot drag the selector around to the far edge.  Its very frustrating.

**Comment by ZMcK** [comment]
Just putting it out there that I'd implement the picker to have an auto converter, so any food that pops up in oz, lbs, etc can be auto converted to grams. FYI, only pushing for this because I use one on the aforementioned app literally all day everyday and it works super well.

**Comment by davidhealey** [comment]
> any food that pops up in oz, lbs,

How would it deal with none standard units. Like 1 biscuit or 1 slice?
